File: C++/exhaustiveSearch.cpp
//Exhaustive Search Algorithm
#include<bits/stdc++.h> 
using namespace std;
float f(float x)
{
    //y=x^2 +54/x;
    return x*x + 54/x;
}
int main()
{
    float a,b,n,x1,x2,x3,d;
    cout<<"Input lower limit and upper limit to find minimum in the interval (a,b) ";
    cin>>a>>b;
    cout<<"Input number of intervals ";
    cin>>n;
    d=(b-a)*1.0/n;
    x1=a;
    x2=x1+d;
    x3=x2+d;
    while(!((f(x1)>=f(x2))&&(f(x2)<f(x3))))
    {
        x1=x2;
        x2=x3;
        x3=x2+d;
    }
    if(x3<=b)cout<<"Minimum lies in the interval "<<x1<<" and "<<x3<<endl;
    else cout<<"No minimum exists in the given interval or the boundary point is the minimum point."<<endl;
}

File: Python/exhaustive_search.py
a, b = map(float, input(
    "Input lower limit, upper limit to find minimum in the interval (a,b): ").split())
choice = int(input(
    "Input 0 to enter number of intervals or 1 to input number of decimal places for accuracy: "))
if(choice == 0):
    n = float(input("Input number of intervals: "))
else:
    t = float(input("Input number of decimal places for accuracy: "))
    n = 2*(b-a)*pow(10, t)
expr = input("Enter expression for x: ")
d = (b-a)/n
x1 = a
x = a
f1 = eval(expr)
x2 = x1+d
x = x2
f2 = eval(expr)
x3 = x2+d
x = x3
f3 = eval(expr)
while(True):
    if(f1 < f2 or f2 > f3):
        x1 = x2
        x = x1
        f1 = eval(expr)
        x2 = x3
        x = x2
        f2 = eval(expr)
        x3 = x2+d
        x = x3
        f3 = eval(expr)
    else:
        break
if(x3 <= b):
    print("Minimum lies in the interval ({},{})".format(
        round(x1, 5), round(x3, 5)))
else:
    print("No minimum exists in the given interval or the boundary point is the minimum point.")
